### PR TITLE
 COMMERCE-1736 sort payment methods base on priority

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by marco-leo** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

File: commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import com.liferay.commerce.payment.result.CommerceSubscriptionStatusResult;
 import com.liferay.commerce.payment.service.CommercePaymentMethodGroupRelLocalService;
 import com.liferay.commerce.payment.util.CommercePaymentUtils;
+import com.liferay.commerce.payment.util.comparator.CommercePaymentMethodPriorityComparator;
```
Comment (marco-leo) [inline commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java:31]: Move This comparator to internal package and avoid version change.

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java (hunk 1 above)
Comment (luca-pellizzon) [reply commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java:31]: @marco-leo CommercePaymentMethodPriorityComparator is a comparator class inside the API module of commerce-payment.
We need it to be exported, we can't put it inside the "internal" package

**Pull request closed without merging**